source-container-build/app/source_build.py

```diff
@@ -173,6 +173,16 @@ def fetch_image_manifest_digest(image: str) -> str:
     return run(cmd, check=True, text=True, capture_output=True).stdout.strip()
 
 
+# produces an artifact name that includes artifact's architecture
+# and repository id in the name
+def unique_srpm_artifact_name(file: str) -> str:
+    root, filename = os.path.split(file)
+    repoid = os.path.basename(root)
+    head, tail = os.path.split(root)
+    arch = os.path.basename(head)
+    return f"{arch}-{repoid}-{filename}"
```

Comment:
The source build script already adds SRPMs gathered from the source container of the base image. Those do not follow this naming scheme, they just have the normal filenames.

IMO we should stick to that as much as possible, only rename if necessary. (It's necessary to rename only if an SRPM with the same filename already exists and isn't the same file)